libyul/Exceptions.h

```diff
@@ -36,7 +36,7 @@ namespace solidity::yul
 struct YulException: virtual util::Exception {};
 struct OptimizerException: virtual YulException {};
 struct CodegenException: virtual YulException {};
-struct YulAssertion: virtual YulException {};
+struct YulAssertion: virtual util::Exception {};
```

Comment:
We should simplify this one day and make all assert macros throw `InternalCompilerError`. Having different types and even documenting them in [Error types](https://docs.soliditylang.org/en/develop/using-the-compiler.html#error-types) just unnecessarily exposes implementation details to the user. We should only have compilation errors there. It's especially confusing when things like `SMTLogicError` are actually failed assertions.